app/src/main/java/com/sdp13epfl2021/projmag/activities/MapsActivity.kt

```diff
@@ -0,0 +1,114 @@
+    }
+
+    private lateinit var mMap: GoogleMap
+    private lateinit var binding: ActivityMapsBinding
+    private var newMarker: Marker? = null
+    private var latitude: Double = Double.MAX_VALUE
+    private var longitude: Double = Double.MAX_VALUE
+    private lateinit var project: ImmutableProject
+    private var nextActivityString: String? = null
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        nextActivityString = intent.getStringExtra(LOCATION_EXTRA)
+
+        project = intent.getParcelableExtra(PROJECT_EXTRA)!!
+        binding = ActivityMapsBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        // Obtain the SupportMapFragment and get notified when the map is ready to be used.
+        val mapFragment = supportFragmentManager
+            .findFragmentById(R.id.map) as SupportMapFragment
+        mapFragment.getMapAsync(this)
+    }
+
+    /**
+     * Manipulates the map once available.
+     * This callback is triggered when the map is ready to be used.
+     * This is where we can add markers or lines, add listeners or move the camera. In this case,
+     * we just add a marker near Sydney, Australia.
+     * If Google Play services is not installed on the device, the user will be prompted to install
+     * it inside the SupportMapFragment. This method will only be triggered once the user has
+     * installed Google Play services and returned to the app.
+     */
+    override fun onMapReady(googleMap: GoogleMap) {
+        mMap = googleMap
+        mMap.moveCamera(CameraUpdateFactory.newLatLngZoom(EPFL, INITIAL_ZOOM))
+
+        if (project.latitude != null && project.longitude != null) {
+            newMarker = mMap.addMarker(
+                MarkerOptions().position(LatLng(project.latitude!!, project.longitude!!))
+                    .title(project.name)
+            )
+        }
+
+        if (nextActivityString == CREATION_STRING) {
+            mMap.setOnMapClickListener { latLng ->
+                newMarker?.let { it.remove() }
+                newMarker = mMap.addMarker(
+                    MarkerOptions().position(latLng).title(
+                        project.name
+                    )
+                )
+                latitude = latLng.latitude
+                longitude = latLng.longitude
+            }
+        }
+
+    }
+
+    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
+        menuInflater.inflate(R.menu.menu_maps_activity, menu)
+        return true
+    }
+
+    override fun onOptionsItemSelected(item: MenuItem): Boolean {
+
+        val nextActivity =
+            if (nextActivityString == CREATION_STRING)
+                ProjectCreationActivity::class.java
+            else
+                ProjectInformationActivity::class.java
+        val newIntent = Intent(this, nextActivity)
+        newIntent.putExtra(PROJECT_EXTRA, project)
+        if (nextActivityString == CREATION_STRING) {
```

Comment:
Maybe doing the same as ProjectCreactionActivity/VideoSubtitlingActivity for exampe, with startActivityForResult/onActivityResult function, would be perhaps better ?
This would also move "if (fromMap) {...}" code outside to the onActivityResult function.